src/main/java/ubc/pavlab/rdp/controllers/UserController.java

```diff
@@ -143,33 +146,101 @@ public ModelAndView faq() {
     }
 
     @GetMapping(value = { "/user/support" })
-    public ModelAndView support() {
+    public ModelAndView support( SupportForm supportForm ) {
         ModelAndView modelAndView = new ModelAndView( "user/support" );
         User user = userService.findCurrentUser();
+        supportForm.setName( user.getProfile().getFullName() );
         modelAndView.addObject( "user", user );
+        modelAndView.addObject( "supportForm", supportForm );
         return modelAndView;
     }
 
+    @Data
+    private static class SupportForm {
+        @NotNull
+        @Size(min = 1, message = "Your must provide your name.")
+        private String name;
+        @NotNull
+        @Size(min = 1, message = "The message must not be empty.")
+        private String message;
+        private MultipartFile attachment;
+    }
+
+    private static class SupportFormValidator implements Validator {
+
+        /**
+         * List of accepted media types as attachment for the support form.
+         */
+        private static final MediaType[] ACCEPTED_MEDIA_TYPES = { MediaType.TEXT_PLAIN, MediaType.IMAGE_JPEG, MediaType.IMAGE_PNG };
+
+        private final String ACCEPTED_MEDIA_TYPES_ERROR_MESSAGE =
+                Arrays.stream( ACCEPTED_MEDIA_TYPES )
+                        .map( MediaType::toString )
+                        .collect( Collectors.joining( ", " ) );
+
+        @Override
+        public boolean supports( Class<?> clazz ) {
+            return SupportForm.class.isAssignableFrom( clazz );
+        }
+
+        @Override
+        public void validate( Object target, Errors errors ) {
+            SupportForm supportForm = (SupportForm) target;
+            MultipartFile multipartFile = supportForm.getAttachment();
+            if ( multipartFile == null || multipartFile.isEmpty() )
+                return;
+            if ( StringUtils.isBlank( multipartFile.getOriginalFilename() ) ) {
+                errors.rejectValue( "attachment", "UserController.SupportFormValidator.attachment.missingFilename" );
+            }
```

Comment:
Maybe make this relative to `UserController.SupportForm` instead of the validator?